coderd/workspaceagents.go

```diff
@@ -461,67 +461,56 @@ func (api *API) enqueueAITaskStateNotification(
 		return
 	}
 
-	workspaceBuild, err := api.Database.GetLatestWorkspaceBuildByWorkspaceID(ctx, workspace.ID)
+	if !workspace.TaskID.Valid {
+		api.Logger.Warn(ctx, "workspace has no task ID")
+		return
+	}
+
+	task, err := api.Database.GetTaskByID(ctx, workspace.TaskID.UUID)
```

Comment:
Review: I wonder if the `subjectNotifier` needs task read permission? Tests didn't complain though 🤔

I guess the fact that we do `AsNotifier` below means this runs as agent, which we amended.

Reply:
I don't believe it does need that permission -- it just needs to be able to enqueue the notification. The agent is doing the work of reading the task data.